doc/cookbook/namedRoutes/NamedRoutes.lhs

```diff
@@ -0,0 +1,369 @@
+# NamedRoutes - Using records to define APIs
+
+Servant offers a very natural way of constructing APIs with nested records, called `NamedRoutes`.
+
+This cookbook explains how to implement such nested-record-APIs using `NamedRoutes` through the example of a Movie Catalog.
+
+First, we start by constructing the domain types of our Movie Catalog.
+After, we show you how to implement the API type with the NamedRoutes records.
+Lastly, we make a Server and a Client out of the API type.
+
+However, it should be understood that this cookbook does _not_ dwell on the built-in servant combinators as the [<Structuring APIs> cookbook ](<https://docs.servant.dev/en/stable/cookbook/structuring-apis/StructuringApis.html>) already covers that angle.
+
+## Why should we use `NamedRoutes` over the alternative `<:|>` operator?
+
+With `NamedRoutes`, we don’t need to care about the declaration order of the endpoints.
+For example, with the `:<|>` operator there’s room for error when the order of the API type
+
+```haskell,ignore
+type API1  =   "version" :> Get '[JSON] Version
+                :<|>  "movies" :> Get '[JSON] [Movie]
+```
+
+does not follow the `Handler` implementation order
+
+```haskell,ignore
+apiHandler :: (AsServerT Handler)
+apiHandler =   getMovies
+                 :<|> getVersion
+```
+
+GHC could scold you with a very tedious message such as :
+
+```console
+    • Couldn't match type 'Handler NoContent'
+                     with 'Movie -> Handler NoContent'
+      Expected type: ServerT MovieCatalogAPI Handler
+        Actual type: Handler Version
+                     :<|> ((Maybe SortBy -> Handler [Movie])
+                           :<|> ((MovieId -> Handler (Maybe Movie))
+                                 :<|> ((MovieId -> Movie -> Handler NoContent)
+                                       :<|> (MovieId -> Handler NoContent))))
+    • In the expression:
+        versionHandler
+          :<|>
+            movieListHandler
+              :<|>
+                getMovieHandler :<|> updateMovieHandler :<|> deleteMovieHandler
+      In an equation for 'server':
+          server
+            = versionHandler
+                :<|>
+                  movieListHandler
+                    :<|>
+                      getMovieHandler :<|> updateMovieHandler :<|> deleteMovieHandler
+    |
+226 | server = versionHandler
+```
+
+On the contrary, with the `NamedRoutes` technique, we refer to the routes by their name:
+
+```haskell,ignore
+data API mode = API
+    { *list*   :: "list" :> ...
```

Comment:
Unfortunately, I don't think the asterisks will put the text in bold within a code block ­— they will likely just appear verbatim.